plugins/catalog-backend-module-gitlab/src/lib/defaultTransformers.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { GroupEntity, UserEntity } from '@backstage/catalog-model';
+import {
+  GitLabGroup,
+  GroupNameTransformerOptions,
+  GroupTransformerOptions,
+  UserTransformerOptions,
+} from './types';
+
+export function defaultGroupNameTransformer(
+  options: GroupNameTransformerOptions,
+): string {
+  if (
+    options.providerConfig.group &&
+    options.group.full_path.startsWith(`${options.providerConfig.group}/`)
+  ) {
+    return options.group.full_path
+      .replace(`${options.providerConfig.group}/`, '')
+      .replaceAll('/', '-');
+  }
+  return options.group.full_path.replaceAll('/', '-');
+}
+
+export function defaultGroupEntitiesTransformer(
+  options: GroupTransformerOptions,
+): GroupEntity[] {
+  const idMapped: { [groupId: number]: GitLabGroup } = {};
+  const entities: GroupEntity[] = [];
+
+  for (const group of options.groups) {
+    idMapped[group.id] = group;
+  }
+
+  for (const group of options.groups) {
+    const annotations: { [annotationName: string]: string } = {};
+
+    annotations[`${options.providerConfig.host}/team-path`] = group.full_path;
```

Comment:
I see this code is just copied from elsewhere so no need to change as part of this PR, but i'll just note for posterity that the namespaces (before the slash) are meant to be hard coded, well known values. Putting a dynamic part there is not how annotations were meant to be used. I had expected this one to be `gitlab.com/team-path` and then possibly the value might be something that contained both host and path if that were necessary, but it seems to me that there's only ever one single of these annotations on each entity so there's probably no need either way to put the host specifically here.

Reply:
+1 on this - annotations are meant to be predictable, maybe we should make an issue about that if we don't want to do it as part of this PR